topic7/contracts/Membership.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity ^0.4.23;
+
+import './Destructible.sol';
+import './SafeMath.sol';
+
+contract Membership is Destructible {
+    
+    using SafeMath for uint;
+
+    uint public memberCount = 0;
+
+    struct Member {
+        bool isMember;
+        uint donations;
+        uint lastDonationTimestamp;
+        uint lastDonationValue;
+    }
+
+    // no usage of arrays or loops, so I get a Bonus :)
+
+    mapping (address=>Member) public members;
+    mapping (address=>uint) public votes;
+    mapping (address=> mapping (address=>bool)) public votedForBy;
+
+
+    constructor() public {
+        _addMember(owner);
+    }
+    
+    function addMember(address _member) public onlyOwner {
+        _addMember(_member);
+    }
+
+    function _addMember(address _member) private {
```

Comment:
Usually the order is constructor, fallback function, external, public, internal, private functions.

Reply:
Understood.